src/scmrepo/git/lfs/fetch.py

```diff
@@ -98,14 +99,35 @@ def get_fetch_url(scm: "Git", remote: Optional[str] = None):  # noqa: C901,PLR09
     return scm.get_remote_url(remote)
 
 
+_FNPATTERN_PREFIX_REGEX = re.compile(r"^(?P<prefix>[^*?\[]*(?:/|$))")
+
+
 def _collect_objects(
     scm: "Git",
     rev: str,
     include: Optional[list[str]],
     exclude: Optional[list[str]],
 ) -> Iterator[Pointer]:
     fs = scm.get_fs(rev)
-    for path in _filter_paths(fs.find("/"), include, exclude):
+    # Optimize path filtering if the `include` list contains exactly one path.
+    # First, determine the root directory wherein to initiate the file search.
+    # If the `include` path is a Unix filename pattern, determine the static
+    # path prefix and set it as the root directory. Second, if the path and the
+    # root are identical or the Unix filename pattern matches *any* (i.e., `**`)
+    # file under the root directory, unset `include` to avoid unnecessary
+    # filtering work.
+    if (
+        include
+        and len(include) == 1
+        and (result := _FNPATTERN_PREFIX_REGEX.match(path := include[0]))
```

Comment:
QQ: what is FNPATTERN (why this name?)

Reply:
okay, I see it's fnmatch. It's fine, super minor - can be probably _ROOT_PATH_PREFIX_REGEX ...

Reply:
Sounds better. :+1: I've amended the commit to use this name.

Reply:
thanks!